share/completions/diskutil.fish

```diff
@@ -8,131 +30,157 @@ function __fish_diskutil_mounted_volumes
     set -l mountpoints /Volumes/*; printf '%s\n' $mountpoints
 end
 
+
+############
+# Commands #
+############
+
 # list
-complete -f -c diskutil -n '__fish_use_subcommand' -a list -d 'List disks'
-complete -f -c diskutil -n '__fish_seen_subcommand_from list' -o 'plist' -d 'Return a property list'
-complete -f -c diskutil -n '__fish_seen_subcommand_from list' -a '(__fish_diskutil_devices)'
+complete -f -c diskutil -n '__fish_diskutil_needs_command' -a list -d 'List disks'
+complete -f -c diskutil -n '__fish_diskutil_using_command list' -o 'plist' -d 'Return a property list'
+complete -f -c diskutil -n '__fish_diskutil_using_command list' -a '(__fish_diskutil_devices)'
```

Comment:
You've probably introduced those two functions above so this won't be triggered for `diskutil apfs list`. If so, just change this condition to `-n '__fish_seen_subcommand_from list; and not __fish_seen_subcommand_from apfs'`.

Or make a wrapper-function that does _that_:

```fish
function __fish_diskutil_using_nonapfs_command
    not __fish_seen_subcommand_from apfs
    and __fish_seen_subcommand_from $argv
end
```

Reply:
How can this be extended to both appleRAID and coreStorage?  They also are subcommands with their own options.

Can I make a function that bundles all three, and change the using command function? E.g.

```
function __fish_diskutil_using_non_subcommand
    not __fish_seen_subcommand_from apfs
    not __fish_seen_subcommand_from appleRAID
    not __fish_seen_subcommand_from coreStorage
    and __fish_seen_subcommand_from $argv
end
```

Reply:
Almost - the second two need to have `and not`:

```fish
function __fish_diskutil_using_not_subcommand
    not __fish_seen_subcommand_from apfs
    and not __fish_seen_subcommand_from appleRAID
    and not __fish_seen_subcommand_from coreStorage
    and __fish_seen_subcommand_from $argv
end
```